src/protocol/metadata.rs

```diff
@@ -4,7 +4,64 @@ use error::Result;
 use codecs::{AsStrings, ToByte, FromByte};
 
 use super::{HeaderRequest, HeaderResponse};
-use super::{API_KEY_METADATA, API_VERSION};
+use super::{API_KEY_API_VERSIONS, API_KEY_METADATA, API_VERSION};
+
+#[derive(Debug)]
+pub struct ApiVersionsRequest<'a> {
+    pub header: HeaderRequest<'a>,
+}
+
+impl<'a> ApiVersionsRequest<'a> {
+    pub fn new(correlation_id: i32, client_id: &'a str) -> ApiVersionsRequest<'a> {
+        ApiVersionsRequest {
+            header: HeaderRequest::new(API_KEY_API_VERSIONS,
+                                       API_VERSION,
```

Comment:
Kafka use the `api_version` for request, it means different request may have different `api_version` on different server version.

For example, [Produce](https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-ProduceAPI) API use 
> v0, v1 (supported in 0.9.0 or later) and v2 (supported in 0.10.0 or later)

And [Offset](https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-OffsetAPI(AKAListOffset)) API use 

> v1 (supported in 0.10.1.0 and later)

So, hard code it will be fine for 0.10 or later, and it always failed on Kafka 0.8